Tasks/MSBuildV1/telemetryHelper.ts

```diff
@@ -0,0 +1,32 @@
+import tl = require('azure-pipelines-task-lib/task');
+import * as semver from 'semver';
+
+export interface TelemetryPayload {
+    msBuildVersion: string;
+    msBuildArguments: string;
+    msBuildLocation: string;
+    msBuildLocationMethod: string;
+    platform: string;
+    configuration: string;
+    msbuildExectionTimeSeconds: number;
+}
+
+export function emitTelemetry(telemetryData: TelemetryPayload) {
+    try {
+        let agentVersion = tl.getVariable('Agent.Version');
+        if (semver.gte(agentVersion, '2.120.0')) {
+            let telemetry = `{"msBuildVersion": "${telemetryData.msBuildVersion}",`;
```

Comment:
Would it be easier to use the object, and convert it once later? It would simplify further changes for telemetry object format (for example, adding nested fields).

Reply:
Reworked this part